index.ts

```diff
@@ -221,6 +224,7 @@ export class CapabilitiesController extends BaseController<any, any> implements
     const methodKey = this.getMethodKeyFor(req.method);
     const permission = this.getPermission(domain.origin, req.method);
     if (methodKey && typeof this.restrictedMethods[methodKey].method === 'function') {
+      const engine = this.createVirtualEngineFor(domain);
```

Comment:
There's some ambiguity here between the `const engine` declared on L227 and the `const engine` declared on L231